server/routerlicious/packages/services-shared/src/socketIoServer.ts

```diff
@@ -77,6 +77,13 @@ class SocketIoServer implements core.IWebSocketServer {
 		this.io.on("connection", (socket: Socket) => {
 			const webSocket = new SocketIoSocket(socket);
 			this.events.emit("connection", webSocket);
+
+			// server side listening for ping events
+			socket.on("ping", (cb) => {
```

Comment:
Your other PR has same changes, not sure if you want to split (and thus remove it from here)